engines/ultima/module.mk

```diff
@@ -0,0 +1,544 @@
+MODULE := engines/ultima
+
+MODULE_OBJS := \
+	detection.o \
+	shared/engine/ultima.o \
+	shared/engine/data_archive.o \
+	shared/engine/debugger.o \
+	shared/engine/events.o \
+	shared/std/misc.o \
+	shared/std/string.o
+
+ifdef ENABLE_ULTIMA1
+MODULE_OBJS += \
+	shared/actions/action.o \
+	shared/actions/huh.o \
+	shared/actions/pass.o \
+	shared/core/base_object.o \
+	shared/core/character.o \
+	shared/core/file.o \
+	shared/core/lzw.o \
+	shared/core/map.o \
+	shared/core/message_target.o \
+	shared/core/mouse_cursor.o \
+	shared/core/named_item.o \
+	shared/core/party.o \
+	shared/core/str.o \
+	shared/core/tree_item.o \
+	shared/core/utils.o \
+	shared/core/widgets.o \
+	shared/early/font_resources.o \
+	shared/early/game.o \
+	shared/early/game_base.o \
+	shared/early/ultima_early.o \
+	shared/engine/debugger.o \
+	shared/engine/input_handler.o \
+	shared/engine/input_translator.o \
+	shared/engine/messages.o \
+	shared/engine/resources.o \
+	shared/gfx/bitmap.o \
+	shared/gfx/character_input.o \
+	shared/gfx/dungeon_surface.o \
+	shared/gfx/font.o \
+	shared/gfx/info.o \
+	shared/gfx/popup.o \
+	shared/gfx/screen.o \
+	shared/gfx/sprites.o \
+	shared/gfx/text_cursor.o \
+	shared/gfx/text_input.o \
+	shared/gfx/viewport_dungeon.o \
+	shared/gfx/viewport_map.o \
+	shared/gfx/visual_container.o \
+	shared/gfx/visual_item.o \
+	shared/gfx/visual_surface.o \
+	shared/maps/map.o \
+	shared/maps/map_base.o \
+	shared/maps/map_tile.o \
+	shared/maps/map_widget.o \
+	shared/maps/creature.o
+endif
+
+ifdef ENABLE_ULTIMA1
```

Comment:
Essentially the first block is for the "shared" code that's currently only used by Ultima 1, but will likely be reused for Ultima 2 and other early games as well. So semantically it just seemed cleaner to have it as a separate block after the shared files that all the sub-engines need, rather than just merging it into the Ultima 1 block.